code/jvm/src/main/kotlin/org/ionproject/codegarten/controllers/api/DeliveriesController.kt

```diff
@@ -1,85 +1,246 @@
+import org.ionproject.codegarten.database.dto.User
+import org.ionproject.codegarten.database.dto.UserClassroom
+import org.ionproject.codegarten.database.dto.UserClassroomMembership.TEACHER
+import org.ionproject.codegarten.database.helpers.DeliveriesDb
+import org.ionproject.codegarten.exceptions.AuthorizationException
+import org.ionproject.codegarten.exceptions.InvalidInputException
+import org.ionproject.codegarten.pipeline.argumentresolvers.Pagination
 import org.ionproject.codegarten.pipeline.interceptors.RequiresUserAuth
-import org.ionproject.codegarten.pipeline.interceptors.USER_ATTRIBUTE
+import org.ionproject.codegarten.pipeline.interceptors.RequiresUserInAssignment
+import org.ionproject.codegarten.remote.github.GitHubInterface
 import org.ionproject.codegarten.responses.Response
+import org.ionproject.codegarten.responses.siren.SirenLink
+import org.ionproject.codegarten.responses.toResponseEntity
+import org.springframework.http.HttpStatus
 import org.springframework.http.ResponseEntity
 import org.springframework.web.bind.annotation.DeleteMapping
 import org.springframework.web.bind.annotation.GetMapping
 import org.springframework.web.bind.annotation.PathVariable
 import org.springframework.web.bind.annotation.PostMapping
 import org.springframework.web.bind.annotation.PutMapping
-import org.springframework.web.bind.annotation.RequestAttribute
-import org.springframework.web.bind.annotation.RequestMapping
-import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RequestBody
 import org.springframework.web.bind.annotation.RestController
+import java.time.OffsetDateTime
+import java.time.format.DateTimeParseException
 
 @RestController
-@RequestMapping("$ORGS_HREF/{orgId}/classrooms/{classroomId}/assignments/{assignmentId}/deliveries")
-class DeliveriesController {
+class DeliveriesController(
+    val deliveriesDb: DeliveriesDb,
+    val gitHub: GitHubInterface
+) {
 
-    @RequiresUserAuth
-    @GetMapping
-    fun getDeliveries(
-        @RequestAttribute(name = USER_ATTRIBUTE) user: UserDao,
-        @PathVariable orgId: Int,
-        @PathVariable classroomId: Int,
-        @PathVariable assignmentId: Int,
-        @RequestParam(defaultValue = PAGE_DEFAULT_VALUE) page: Int,
-        @RequestParam(defaultValue = COUNT_DEFAULT_VALUE) count: Int
+    @RequiresUserInAssignment
+    @GetMapping(DELIVERIES_HREF)
+    fun getAllDeliveries(
+        @PathVariable(name = ORG_PARAM) orgId: Int,
+        @PathVariable(name = CLASSROOM_PARAM) classroomNumber: Int,
+        @PathVariable(name = ASSIGNMENT_PARAM) assignmentNumber: Int,
+        pagination: Pagination,
+        user: User,
+        userClassroom: UserClassroom,
+        assignment: Assignment
     ): ResponseEntity<Response> {
-        TODO()
+        val deliveries = deliveriesDb.getDeliveriesOfAssignment(orgId, classroomNumber, assignmentNumber, pagination.page, pagination.limit)
```

Comment:
- Can't these two things be resolved in the same DB interaction?
- Do we really need the count?

Reply:
The "count" is used to obtain the total amount of, in this case, deliveries. This information is useful to know how many items exist in total, since the results can be paginated. It's probably possible to obtain the "count" in the same interaction where we obtain the result set, which can be a possible improvement in the future.